doc/languages-frameworks/go.section.md

```diff
@@ -62,6 +62,62 @@ The following is an example expression using `buildGoModule`:
 }
 ```
 
+### Obtaining and overriding `vendorHash` for `buildGoModule` {#buildGoModule-vendorHash}
+
+`nix-prefetch` can be used to obtain the actual hash. The following command gets the value of `vendorHash` for package `pet`:
+
+```sh
+cd path/to/nixpkgs
+nix-prefetch -E "{ hash }: ((import ./. { }).my-package.overrideAttrs (_: { vendorHash = hash; })).go-modules"
+```
+
+To obtain the hash without external tools, set `vendorHash = lib.fakeHash;` and run the build. ([more details here](#sec-source-hashes)).
+
+`vendorHash` can be overridden with `overrideAttrs`. Override the above example like this:
+
+```nix
+{
+  pet_0_4_0 = pet.overrideAttrs (
+    finalAttrs: previousAttrs: {
+      version = "0.4.0";
+      src = fetchFromGitHub {
+        inherit (previousAttrs.src) owner repo;
+        rev = "v${finalAttrs.version}";
+        hash = "sha256-gVTpzmXekQxGMucDKskGi+e+34nJwwsXwvQTjRO6Gdg=";
+      };
+      vendorHash = "sha256-dUvp7FEW09V0xMuhewPGw3TuAic/sD7xyXEYviZ2Ivs=";
+    }
+  );
+}
+```
+
+### Overriding `goModules` (#buildGoModule-goModules-override)
+
+Overriding `<pkg>.goModules` by calling `goModules.overrideAttrs` is unsupported. Still, it is possible to override the `vendorHash` (`goModules`'s `outputHash`) and the `pre`/`post` hooks for both the build and patch phases of the primary and `goModules` derivation. Alternatively, the primary derivation provides an overridable `passthru.overrideModAttrs` function to store the attribute overlay implicitly taken by `goModules.overrideAttrs`. Here's an example usage of `overrideModAttrs`:
+
+```nix
+let
+  pet-foo = pkgs.overrideAttrs (
+    finalAttrs: previousAttrs: {
+      passthru = previousAttrs.passthru // {
+        overrideModAttrs = lib.composeExtensions previousAttrs.passthru.overrideModAttrs (
+          finalModAttrs: previousModAttrs: {
+            # goModules-specific overriding goes here
+            postBuild = ''
+              # Dynamically patche the fetched files
+              substituteInPlace vendor/github.com/example/repo/file.go \
+                --replace-fail "panic(err)" ""
+            '';
+          }
+        );
+      };
+    }
+  );
+  pet-vendored = pet-foo.overrideAttrs { vendorHash = null; };
```

Comment:
This is a bit surprising, perhaps the paragraph above should mention why you set `vendorHash = null;`? Or perhaps explain this here in a a comment? (I think I understood correctly this is because you set `overrideModAttrs`...)

Reply:
`venhorHash = null` is to prevent vendoring from the Nix side by `goModules`. People set it because the source is "vendored" (contains the `vendor` directory), and they would like to use such vendoring. For example, the Apptainer project provided vendored source tarballs in each release when the project was still known as Singularity.

Reply:
> `venhorHash = null` is to prevent vendoring from the Nix side by `goModules`. People set it because the source is "vendored" (contains the `vendor` directory), and they would like to use such vendoring. For example, the Apptainer project provided vendored source tarballs in each release when the project was still known as Singularity.
> 
> If a user would like to use the `goModules` vendoring instead of the upstream one, they would set `deleteVendor = true` instead.

I'm sorry but I did not understand :). I'll write again what I wanted you to explain in a comment / what I felt is missing:

The purpose of this example is to demonstrate how to use `overrideModAttrs`, but what is peculiar to me, is that you perform an additional `overrideAttrs` of `vendorHash = null;`. I don't understand how that is related to `overrdeModAttrs` - if you set `overrideModAttrs` do you have to also set `vendorHash = null;` afterwards? If so, why (explain in a code comment)? If you don't have to set `vendorHash = null;` afterwards, please remove that, and finish with:

```nix
in 
pet-foo
```

Reply:
The first overriding targeting `goModules` might be buried somewhere inside the codebase, while another user made another overriding and set `vendorHash = null`.

The purpose of this documentation is to avoid the conflicts between these overridings.